stumpy/Mmotifs.py

```diff
@@ -0,0 +1,297 @@
+import numpy as np
+import logging
+import stumpy
+import matplotlib.pyplot as plt
+from . import core
+from . import config
+from stumpy.core import _preprocess
+
+logger = logging.getLogger(__name__)
+
+
+def Mmotifs(
+    T: np.ndarray,
+    P: np.ndarray,
+    I: np.ndarray,
+    max_matches: int = 10,
+    max_motifs: int = 1,
+    cutoff: np.ndarray = None,
+    visualize_mdl: bool = False,
+    max_distance: float = None,
+    min_neighbors: int = 1,
+    atol: float = 1e-8,
+):
+    """
+    Discover the top k multidimensional motifs for the time series T
+
+    Parameters
+    ----------
+    T: numpy.ndarray
+        The multidimensional time series or sequence
+
+    P: numpy.ndarray
+        Multidimensional Matrix Profile of T
+
+    I: numpy.ndarray
+        Matrix Profile indices
+
+    max_matches: int, default 10
+        The maximum amount of similar matches (nearest neighbors) of a motif
+        representative to be returned.
+        The first match is always the self-match for each motif.
+
+    max_distance: flaot, default None
+        Maximal distance that is allowed between a query subsequence
+        (a candidate motif) and all subsequences in T to be considered as a match.
+        If None, this defaults to
+        `np.nanmax([np.nanmean(D) - 2 * np.nanstd(D), np.nanmin(D)])`
+        (i.e. at least the closest match will be returned).
+
+    min_neighbors : int, default 1
+        The minimum number of similar matches a subsequence needs to have in order
+        to be considered a motif. This defaults to `1`, which means that a subsequence
+        must have at least one similar match in order to be considered a motif.
+
+    atol : float, default 1e-8
+        The absolute tolerance parameter. This value will be added to `max_distance`
+        when comparing distances between subsequences.
+
+    max_motifs: int, default 1
+        The maximum number of motifs to return
+
+    cutoff: numpy.ndarray or float, default None
+    The largest matrix profile value (distance) for each dimension of the
+    multidimensional matrix profile that a multidimenisonal candidate motif is
+    allowed to have.
+    If cutoff is only one value, these value will be applied to every dimension.
+

```

Comment:
Can we please make sure that the order of these parameters match the order that is found in `motifs()`?